processor/src/main/java/io/norberg/automatter/processor/Descriptor.java

```diff
@@ -224,10 +224,17 @@ private List<ExecutableElement> methods(final TypeElement element) {
 
   private void enumerateMethods(
       final TypeElement element, final Map<String, ExecutableElement> methods) {
-    for (final TypeMirror interfaceType : element.getInterfaces()) {
-      final TypeElement interfaceElement = (TypeElement) ((DeclaredType) interfaceType).asElement();
-      enumerateMethods(interfaceElement, methods);
+    if (isRecord) {
```

Comment:
This still doesn't seem to reliably give us the fields in the order that the record constructor expects them.

To see this, try permutating the order of fields in `BazRecord`. It'll still satisfy the interfaces but now the build fails to compile.

Reply:
Which order can I try to reproduce it? I tried different ones but haven't found one.

Reply:
Ah no, I was just confused, this always works because `enumerateEnclosedMethods` returns all methods in the same order that the record declares the fields.

Reply:
👍